src/main/java/com/qouteall/immersive_portals/commands/MyCommandServer.java

```diff
@@ -888,43 +905,124 @@ private static void setMultidestEntry(
                 },
                 Portal.entityType
             );
-        }
-        else if (biFaced) {
+        } else if (biFaced) {
             PortalManipulation.completeBiFacedPortal(newPortal, Portal.entityType);
-        }
-        else if (biWay) {
+        } else if (biWay) {
             PortalManipulation.completeBiWayPortal(newPortal, Portal.entityType);
         }
     }
-    
+
     public static void sendPortalInfo(CommandContext<ServerCommandSource> context, Portal portal) {
         context.getSource().sendFeedback(
             portal.toTag(new CompoundTag()).toText(),
             false
         );
-        
+
         sendMessage(
             context,
             "\n\n" + portal.toString()
         );
     }
-    
+
     public static void reloadPortal(Portal portal) {
         portal.updateCache();
         McHelper.getIEStorage(portal.dimension).resendSpawnPacketToTrackers(portal);
     }
-    
+
     public static void sendMessage(CommandContext<ServerCommandSource> context, String message) {
         context.getSource().sendFeedback(
             new LiteralText(message),
             false
         );
     }
-    
+
+    // By LoganDark :D
+    private static int runMakePortal(CommandContext<ServerCommandSource> context) {
+        try {
+            double width = DoubleArgumentType.getDouble(context, "width");
+            double height = DoubleArgumentType.getDouble(context, "height");
+            DimensionType to = DimensionArgumentType.getDimensionArgument(context, "to");
+            Vec3d dest = Vec3ArgumentType.getVec3(context, "dest");
+
+            ServerPlayerEntity player = context.getSource().getPlayer();
+            Vec3d playerLook = player.getRotationVector();
+
+            net.minecraft.util.Pair<BlockHitResult, List<Portal>> rayTrace =
+                Helper.rayTrace(
+                    player.world,
+                    new RayTraceContext(
+                        player.getCameraPosVec(1.0f),
+                        player.getCameraPosVec(1.0f).add(playerLook.multiply(100.0)),
+                        RayTraceContext.ShapeType.OUTLINE,
+                        RayTraceContext.FluidHandling.NONE,
+                        player
+                    ),
+                    true
+                );
+
+            BlockHitResult hitResult = rayTrace.getLeft();
+            List<Portal> hitPortals = rayTrace.getRight();
+
+            if (Helper.hitResultIsMissedOrNull(hitResult)) {
+                return 0;
+            }
+
+            AtomicReference<Vec3d> playerLookAtomic = new AtomicReference<>(playerLook);
+            hitPortals.forEach(portal -> playerLookAtomic.set(portal.transformLocalVec(playerLookAtomic.get())));
```

Comment:
Can be changed to reduce

Reply:
I was thinking the same thing but Java doesn't have an easy reduce. I'll change this to a foreach loop